src/components/carousel/index.tsx

```diff
@@ -46,7 +54,9 @@ const Carousel: FC<Props> = (props) => {
       fallback: false,
     }
   );
-  const hasPagination = fullScreen && !isSmallLandscapeViewport;
+
+  const hasThumbnailPagination = fullScreen && !isSmallLandscapeViewport;
+  const hasNumbersPagination = withNumbersPagination && !fullScreen;
```

Comment:
at the moment those are mutually exclusive because of the fullscreen prop. But if someone changes the first condition, you could end up with an impossible state where both are true.
wdyt about this:
```typescript
let paginationType = "none";
if (fullScreen) {
  paginationType = isSmallLandscapeViewport ? "none" : "thumbnail";
} else {
  paginationType = withNumbersPagination ? "number" : "none";
}
```

Reply:
usually you always want to prefer a type/status variable instead of booleans because of this problem. Here is a good article: https://kentcdodds.com/blog/stop-using-isloading-booleans

Reply:
Out of scope, just some thoughts:

Same goes to the interface of the component itself, imagine as a consumer of the component you pass `fullScreen={true} withNumbersPagination={true}`. Without knowing the internals, I would expect the pagination to be shown. But it isn't. It is out of scope for this ticket but what we could do in a follow up is something like a CarouselVariant prop that would move the condition out of the component and leave it to the consumer.

Reply:
Thanks very good points. So, the thing is that you could have numbers pagination on normal mode, but on full screen it will be thumbnails. We are not expecting to use numbers on a full screen mode.